### PR TITLE
Add wheel builds

### PR DESCRIPTION
**Pull request by malfet**

### CODE REVIEW

File: .github/workflows/wheels.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        uses: actions/checkout@v2
+      - name: Install PyTorch Nightly
+        run: |
+          python3 -mpip install --pre torch -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
+      - name: Build wheel
+        run: |
+          python3 -mpip install wheel
+          python3 setup.py bdist_wheel
+      - name: Upload wheel as GHA artifact
+        uses: actions/upload-artifact@v2
```
Comment (zou3519) [inline .github/workflows/wheels.yml:31]: @malfet where does this end up uploading to?

File: .github/workflows/wheels.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+on:
+  push:
+    branches:
+      - main
+  pull_request
+
+jobs:
+
+  build-wheel:
+    runs-on: ubuntu-18.04
```
Comment (zou3519) [inline .github/workflows/wheels.yml:11]: Do you have guidance for how to do builds on Mac or Windows?

File: .github/workflows/wheels.yml (hunk 1 above)
Comment (malfet) [reply .github/workflows/wheels.yml:31]: To projects actions page, for example https://github.com/pytorch/functorch/actions/runs/1604493500

File: .github/workflows/wheels.yml (hunk 2 above)
Comment (malfet) [reply .github/workflows/wheels.yml:11]: Let me extend the matrix to Mac, not sure about Windows though

Comment (malfet) [reply .github/workflows/wheels.yml]: Done

File: .github/workflows/wheels.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+name: Wheels
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches:
+      - main
```
Comment (zou3519) [inline .github/workflows/wheels.yml:7]: I think this file makes sense to me. When we are ready to cut a release binary against PyTorch 1.11, would we create a separate functorch branch (e.g. release/0.1.0) and modify this file to install the PyTorch RC?

**Review by zou3519 — APPROVED** [review]
This is pretty cool, thanks @malfet!

Looking at the CI signal you might need to rebase this PR on top of the main branch

File: .github/workflows/wheels.yml (hunk 3 above)
Comment (malfet) [reply .github/workflows/wheels.yml:7]: Good point, let me add `- release/*` wildcard, and tags one as well

File: .github/workflows/wheels.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+name: Wheels
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
```
Comment (vfdev-5) [inline .github/workflows/wheels.yml:4]: @zou3519 why we need to execute this workflow (upload wheels) on each PR ? Maybe I'm missing something ...

Comment (zou3519) [reply .github/workflows/wheels.yml:4]: I think the upload wheels only uploads the wheels to github actions. We technically don't need to do this on each PR if we want to save bandwidth but it is still a good check that the wheel building command runs without problem

**Pull request merged**